translate/storage/test_mozilla_lang.py

```diff
@@ -121,3 +127,87 @@ def test_ok_translations(self, ok, target, istranslated):
         assert unit.source == "Source"
         assert unit.target == target
         assert unit.istranslated() == istranslated
+
+    def test_headers(self):
+        """Ensure we can handle and preserve file headers"""
+        lang = ("## active ##\n"
+                "## some_tag ##\n"
+                "## another_tag ##\n"
+                "## NOTE: foo\n"
+                ";Source\n"
+                "Target\n"
+                "\n\n")
+        store = self.StoreClass.parsestring(lang)
+        assert (
+            store.getlangheaders()
+            == [u'## some_tag ##',
+                u'## another_tag ##',
+                u'## NOTE: foo'])
+        out = io.BytesIO()
+        store.serialize(out)
+        out.seek(0)
+        assert (
+            out.read()
+            == ("## active ##\n"
+                "## some_tag ##\n"
+                "## another_tag ##\n"
+                "## NOTE: foo\n"
+                "\n\n"
+                ";Source\n"
+                "Target\n"
+                "\n\n"))
+
+    def test_not_headers(self):
+        """Ensure we dont treat a tag immediately after headers as header"""
+        lang = ("## active ##\n"
+                "## some_tag ##\n"
+                "## another_tag ##\n"
+                "## NOTE: foo\n"
+                "## TAG: fooled_you ##\n"
+                ";Source\n"
+                "Target\n"
+                "\n\n")
+        store = self.StoreClass.parsestring(lang)
+        assert "## TAG: fooled_you ##" not in store.getlangheaders()
+
+    def test_tag_comments(self):
+        """Ensure we can handle comments and distinguish from headers"""
+        lang = ("## active ##\n"
+                "# First comment\n"
+                "## TAG: important_tag\n"
+                "# Second comment\n"
+                "# Third comment\n"
+                "## TAG: another_important_tag\n"
+                ";Source\n"
+                "Target\n"
+                "\n\n")
+        store = self.StoreClass.parsestring(lang)
+        assert not store.getlangheaders()
+        assert bytes(store).decode('utf-8') == lang
+        assert (
+            "# TAG: important_tag"
+            in store.units[0].getnotes(origin="developer").split("\n"))
+        lang = ("## active ##\n"
+                "# First comment\n"
+                "## TAG: important_tag\n"
+                "# Second comment\n"
+                "# Third comment\n"
+                "## TAG: another_important_tag\n"
+                "# Another comment\n"
+                ";Source\n"
+                "Target\n"
+                "\n\n")
+        store = self.StoreClass.parsestring(lang)
+        assert not store.getlangheaders()
+        assert (
+            "First comment"
+            in store.units[0].getnotes(origin="developer").split("\n"))
```

Comment:
Is it not better to get all the dev notes for the first units and store in variable? Considerable shorter asserts I would say.